tests/conftest.py

```diff
@@ -269,6 +284,99 @@ def global_sa_results_dir(datadir: Path) -> Path:
     return result_dir
 
 
+@pytest.fixture()
+def up_results_dir(datadir: Path) -> Path:
+    """
+    Crete a uncertainty propagation analyses result folder
+    with a result file.
+    """
+
+    import numpy as np
+
+    result_dir = datadir / "fake_dir"
+    result_dir.mkdir(parents=True)
+    time_set = np.array([0, 1, 2, 3, 4, 5, 6, 7])
+    realization_outputs = np.array(
+        [
+            [
+                [
+                    i + (qoi_index * 10) + tl
+                    for tl in (0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7)
+                ]
+                for qoi_index, trend_property in enumerate(
+                    [("trend_id_1", "temperature"), ("trend_id_1", "absolute_pressure")]
+                )
+            ]
+            for i in range(5)
+        ]
+    )
```

Comment:
This is probably a case where nested `for`s would be better to read/understand than list comprehensions. Also I recommend replacing vars `i` and `tl` by real words here to make it clear what they represents.

Reply:
I disagree, particularly I find that list comprehension quite readable.

About the variable names I think it is up to the caller, but in list comprehensions I'm at least used to short named variables: given list comprehensions have a very narrow scope, single letter-vars in there are considered OK.